feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -799,114 +799,110 @@ void apc_state_set(struct blob_attr *msg)
 		}
 	}
 
-	LOGD("APC_state Updating");
+	LOGI("APC_state Updating: mode: %s, dr-addr: %s bdr-addr: %s", 
+	     apc_state.mode, apc_state.dr_addr, apc_state.bdr_addr);
+
 	if (!ovsdb_table_update(&table_APC_State, &apc_state))
 		LOG(ERR, "APC_state: failed to update");
+
 }
 
 static ovsdb_table_t table_Manager;
-ev_timer apc_cld_mon_timer;
-ev_timer apc_cld_disconn_timer;
-ev_timer apc_backoff_timer;
-#define APC_CLD_MON_TIME 5
-#define APC_CLD_DISCONN_TIME 60
-#define APC_BKOFF_TIME 60
+static int conn_since = 0;
+#define APC_CLOUD_MON_PERIOD 60
+
+static void apc_enable(bool flag) {
+
+	SCHEMA_SET_INT(apc_conf.enabled, flag);
+	if (!ovsdb_table_update(&table_APC_Config, &apc_conf)) {
+		LOG(ERR, "%s:APC_Config: failed to update", __func__);
+		return;
+	}
+	LOGI("APC %s: %s APC", __func__, flag?"enable":"disable");
+
+}
 
 static void
-apc_cld_mon_cb(struct ev_loop *loop, ev_timer *timer, int revents)
+apc_cld_mon_cb(struct schema_Manager *mgr)
 {
-	struct schema_APC_State state;
-	struct schema_Manager mgr;
+	int i = 0;
+	conn_since = 0;
+	struct schema_APC_State apc_state;
 	json_t *where;
+	int ret = 0;
+	int link = 1;
 
-	where = ovsdb_table_where(&table_APC_State, &state);
+	where = ovsdb_table_where(&table_APC_State, &apc_state);
 	if (false == ovsdb_table_select_one_where(&table_APC_State,
-			where, &state)) {
+						  where, &apc_state)) {
 		LOG(ERR, "%s: APC_State read failed", __func__);
 		return;
 	}
-	/*If not DR and not BDR the dont proceed */
-	if ((strncmp(state.mode, "DR", 2) != 0) &&
-	    (strncmp(state.mode, "BDR", 3) != 0))
-		return;
-
-	/*If BDR addr is 0 means its a standalone DR*/
-	if ((strncmp(state.bdr_addr, "0.0.0.0", 7) == 0))
-		return;
 
-	where = ovsdb_table_where(&table_Manager, &mgr);
-	if (false == ovsdb_table_select_one_where(&table_Manager,
-			where, &mgr)) {
-		LOG(ERR, "%s: Manager read failed", __func__);
-		return;
+	for(i=0; i < mgr->status_len; i++) {
+		if(!strncmp(mgr->status_keys[i] , "sec_since_connect", 17 ))
+			conn_since = atoi(mgr->status[i]);
```

Comment:
Again, no functionality change, but can use strlen("sec_since_connect") instead of hard coded 17.
Did you want to break once you get the connection time info from the manager?

Reply:
You might also want to bring this block of code down after checking for the Manager connection status. Or, inside the "else" block where you are sure that the Manager is connected.